adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java

```diff
@@ -0,0 +1,400 @@
+     * 
+     * @param response response
+     * @param msg message to write in the response
+     */
+    public static void badRequest(final HttpServerResponse response, final String msg) {
+        response.setStatusCode(HTTP_BAD_REQUEST).end(msg);
+    }
+
+    /**
+     * Ends the given response with status code HTTP_INTERNAL_ERROR and the given message.
+     *
+     * @param response response
+     * @param msg message to write in the response
+     */
+    public static void internalServerError(final HttpServerResponse response, final String msg) {
+        response
+            .setStatusCode(HTTP_INTERNAL_ERROR)
+            .putHeader(HttpHeaders.CONTENT_TYPE, "text/plain; charset=utf-8")
+            .end(msg);
+    }
+
+    /**
+     * Ends the given response with status code HTTP_UNAVAILABLE and sets a RETRY_AFTER header with the given number of seconds.
+     * 
+     * @param response response
+     * @param retryAfterSeconds number of seconds to set in the RETRY_AFTER header
+     */
+    public static void serviceUnavailable(final HttpServerResponse response, final int retryAfterSeconds) {
+        serviceUnavailable(response, retryAfterSeconds, null, null);
+    }
+
+    /**
+     * Ends the given response with status code HTTP_UNAVAILABLE and sets a RETRY_AFTER header with the given number of seconds.
+     * Additionally sets the given message and content type.
+     * 
+     * @param response response
+     * @param retryAfterSeconds number of seconds to set in the RETRY_AFTER header
+     * @param msg message to write in the response
+     * @param contentType content type of the message
+     */
+    public static void serviceUnavailable(final HttpServerResponse response, final int retryAfterSeconds,
+            final String msg, final String contentType) {
+        response
+            .setStatusCode(HTTP_UNAVAILABLE)
+            .putHeader(HttpHeaders.RETRY_AFTER, String.valueOf(retryAfterSeconds));
+        if (msg != null) {
+            response
+                .putHeader(HttpHeaders.CONTENT_TYPE, contentType)
+                .end(msg);
+        } else {
+            response.end();
+        }
+    }
+
+    private void doGetStatus(final RoutingContext ctx) {
+        JsonObject result = new JsonObject(hono.getConnectionStatus());
+        result.put("active profiles", activeProfiles);
+        result.put("senders", hono.getSenderStatus());
+        adaptStatusResource(result);
+        ctx.response()
+            .putHeader(HttpHeaders.CONTENT_TYPE, CONTENT_TYPE_JSON)
+            .end(result.encodePrettily());
+    }
+
+    /**
+     * Adapts the JsonObject returned on a status request.
+     * Subclasses can add their own properties here.
+     * 
+     * @param status status object to be adapted
+     */
+    protected void adaptStatusResource(JsonObject status) {
+        // empty
+    }
+
+    public static String getContentType(final RoutingContext ctx) {
```

Comment:
if the idea is to provide a base class to be subclassed, then this (and the following) helper methods should probably be declared *protected*, shouldn't they?